tools/util.py

```diff
@@ -374,3 +373,120 @@ def mkdtemp():
     # 'TS5009: Cannot find the common subdirectory path for the input files.'
     temp_dir = os.environ["TEMP"] if os.name == 'nt' else None
     return tempfile.mkdtemp(dir=temp_dir)
+
+
+class DenoTestCase(unittest.TestCase):
+    @property
+    def build_dir(self):
+        args = test_args()
+        return args.build_dir
```

Comment:
@hayd can you cache this? No need to parse args every time this property is accessed

Reply:
This is quite fiddly (it can be cached but it feels like duct tape). I'll need to think about this a little bit.

Reply:
How about:
```python
    @property
    def build_dir(self):
        if hasattr(self, "_cached_build_dir"):
            return self._cached_build_dir
        args = test_args()
        # cache for subsequent use
        self._cached_build_dir = args.build_dir
        return self._cached_build_dir
```